pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -228,8 +255,69 @@ func (r *ReconcileIBMBlockCSI) Reconcile(request reconcile.Request) (reconcile.R
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileIBMBlockCSI) addFinalizerIfNotPresent(instance *ibmblockcsi.IBMBlockCSI) error {
+	logger := log.WithName("addFinalizer")
+
+	accessor, finalizerName, err := r.getAccessorAndFinalizerName(instance)
+	if err != nil {
+		return err
+	}
+
+	if !util.Contains(accessor.GetFinalizers(), finalizerName) {
+		logger.Info("adding", "finalizer", finalizerName, "on", accessor.GetName())
+		accessor.SetFinalizers(append(accessor.GetFinalizers(), finalizerName))
+
+		if err := r.client.Update(context.TODO(), instance.Unwrap()); err != nil {
+			logger.Error(err, "failed to add", "finalizer", finalizerName, "on", accessor.GetName())
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileIBMBlockCSI) hasFinalizer(instance *ibmblockcsi.IBMBlockCSI) (bool, error) {
+	accessor, finalizerName, err := r.getAccessorAndFinalizerName(instance)
+	if err != nil {
+		return false, err
+	}
+
+	if util.Contains(accessor.GetFinalizers(), finalizerName) {
+		return true, nil
+	}
+	return false, nil
```

Comment:
Can we merge these lines together to:
```suggestion
	return util.Contains(accessor.GetFinalizers(), finalizerName), nil
```
?
The function name and this code are pretty clear even without this `if` statement.

Reply:
done